classes/solver.rb

```diff
@@ -0,0 +1,24 @@
+class Solver
+  def factorial(num, product = 1)
+    raise ArgumentError, 'Negative numbers are not allowed' if num.negative?
+    return product if [1, 0].include?(num)
+
+    factorial(num - 1, product * num)
+  end
+
+  def reverse_string(str)
+    str.chars.reverse.join
+  end
+
+  def fizzbuzz(num)
+    if (num % 3).zero? && (num % 5).zero?
+      'fizzbuzz'
+    elsif (num % 3).zero?
+      'fizz'
+    elsif (num % 5).zero?
+      'buzz'
+    else
+      num.to_s
+    end
+  end
+end
```

Comment:
According to the project [requirements](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md), " `Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.` ", please make sure you write the tests first before implementing the functions, and that should be reflected in your commit history.